src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IDelegateCreationExpression.cs

```diff
@@ -0,0 +1,1418 @@
+        {
+            string source = @"
+class Program
+{
+    delegate void DType();
+    void Main()
+    {
+        DType /*<bind>*/d1 = () =>/*</bind>*/;
+    }
+}
+";
+            string expectedOperationTree = @"
+IVariableDeclarationStatement (1 declarations) (OperationKind.VariableDeclarationStatement, IsInvalid) (Syntax: 'DType /*<bi ... *</bind>*/;')
+  IVariableDeclaration (1 variables) (OperationKind.VariableDeclaration, IsInvalid) (Syntax: 'DType /*<bi ... *</bind>*/;')
+    Variables: Local_1: Program.DType d1
+    Initializer: IDelegateCreationExpression (OperationKind.DelegateCreationExpression, Type: Program.DType, IsInvalid) (Syntax: '() =>/*</bind>*/')
+        Target: IAnonymousFunctionExpression (Symbol: lambda expression) (OperationKind.AnonymousFunctionExpression, Type: null, IsInvalid) (Syntax: '() =>/*</bind>*/')
+            IBlockStatement (2 statements) (OperationKind.BlockStatement, IsInvalid) (Syntax: '')
+              IExpressionStatement (OperationKind.ExpressionStatement, IsInvalid) (Syntax: '')
+                Expression: IInvalidExpression (OperationKind.InvalidExpression, Type: null, IsInvalid) (Syntax: '')
+                    Children(0)
+              IReturnStatement (OperationKind.ReturnStatement, IsInvalid) (Syntax: '')
+                ReturnedValue: null
+";
+            var expectedDiagnostics = new DiagnosticDescription[] {
+                // CS1525: Invalid expression term ';'
+                //         DType /*<bind>*/d1 = () =>/*</bind>*/;
+                Diagnostic(ErrorCode.ERR_InvalidExprTerm, ";").WithArguments(";").WithLocation(7, 46)
+            };
+
+            VerifyOperationTreeAndDiagnosticsForTest<VariableDeclaratorSyntax>(source, expectedOperationTree, expectedDiagnostics,
+                AdditionalOperationTreeVerifier: new ExpectedSymbolVerifier().Verify);
+        }
+
+        [CompilerTrait(CompilerFeature.IOperation)]
+        [Fact]
+        public void DelegateCreationExpression_ImplicitMethodBinding_MultipleCandidates_InvalidNoMatch()
+        {
+            string source = @"
+using System;
+class Program
+{
+    void Main()
+    {
+        /*<bind>*/Action<int> a = M1;/*</bind>*/
+    }
+    void M1(object o) { }
+    void M1(string s) { }
+}
+";
+            string expectedOperationTree = @"
+IVariableDeclarationStatement (1 declarations) (OperationKind.VariableDeclarationStatement, IsInvalid) (Syntax: 'Action<int> a = M1;')
+  IVariableDeclaration (1 variables) (OperationKind.VariableDeclaration, IsInvalid) (Syntax: 'Action<int> a = M1;')
+    Variables: Local_1: System.Action<System.Int32> a
+    Initializer: IDelegateCreationExpression (OperationKind.DelegateCreationExpression, Type: System.Action<System.Int32>, IsInvalid) (Syntax: 'M1')
+        Target: IMethodBindingExpression: void Program.M1(System.Object o) (OperationKind.MethodBindingExpression, Type: null, IsInvalid) (Syntax: 'M1')
```

Comment:
>Target: IMethodBindingExpression: void Program.M1(System.Object o) (OperationKind.MethodBindingExpression, Type: null, IsInvalid) (Syntax: 'M1') [](start = 8, length = 144)

This feels incorrect, the method isn't a match.

Reply:
The issue here was that the BoundMethodGroup itself is fine, but the conversion sitting on top of it isn't. To resolve, I'm filtering the method group when created. If the conversion sitting on top `HasErrors`, then I fall back to the CSharpOperationFactory, which returns a NoneOperation here.

---
In reply to: [140108124](https://github.com/dotnet/roslyn/pull/22056#discussion_r140108124) [](ancestors = 140108124)